src/tools/spake2p/Cmd_GenVerifier.cpp

```diff
@@ -0,0 +1,259 @@
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements the command handler for the 'spake2p' tool
+ *      that generates Verifier.
+ *
+ */
+
+#ifndef __STDC_LIMIT_MACROS
+#define __STDC_LIMIT_MACROS
+#endif
+
+#include "spake2p.h"
+
+#include <CHIPVersion.h>
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/support/Base64.h>
+#include <lib/support/CHIPArgParser.hpp>
+#include <lib/support/CHIPMem.h>
+#include <protocols/secure_channel/PASESession.h>
+
+namespace {
+
+using namespace chip::ArgParser;
+
+#define CMD_NAME "spake2p gen-verifier"
+
+bool HandleOption(const char * progName, OptionSet * optSet, int id, const char * name, const char * arg);
+
+// clang-format off
+OptionDef gCmdOptionDefs[] =
+{
+    { "count",           kArgumentRequired, 'c' },
+    { "pin-code",        kArgumentRequired, 'p' },
+    { "iteration-count", kArgumentRequired, 'i' },
+    { "salt",            kArgumentRequired, 's' },
+    { "out",             kArgumentRequired, 'o' },
+    { }
+};
+
+const char * const gCmdOptionHelp =
+    "   -c, --count <int>\n"
+    "\n"
+    "       The number of pin-code/verifier parameter sets to be generated. If not specified,\n"
+    "       one set will be generated.\n"
+    "\n"
+    "   -p, --pin-code <int>\n"
+    "\n"
+    "       SPAKE2P setup PIN code. The value should be positive integer in range [1..99999998].\n"
+    "       If not specified, the PIN code value will be randomly generated.\n"
+    "       When count is more than one, only first set will use the specified PIN code value\n"
+    "       and others will be randomly generated.\n"
+    "       The following PIN codes SHALL NOT be used due to their trivial, insecure nature:\n"
+    "          * 00000000\n"
+    "          * 11111111\n"
+    "          * 22222222\n"
+    "          * 33333333\n"
+    "          * 44444444\n"
+    "          * 55555555\n"
+    "          * 66666666\n"
+    "          * 77777777\n"
+    "          * 88888888\n"
+    "          * 99999999\n"
+    "          * 12345678\n"
+    "          * 87654321\n"
+    "\n"
+    "   -i, --iteration-count <int>\n"
+    "\n"
+    "       SPAKE2P PBKDF iteration count. The value should be positive integer in range [1000..100000].\n"
+    "\n"
+    "   -s, --salt <string>\n"
+    "\n"
+    "       SPAKE2P PBKDF salt input value. Legth of salt string should be in range [16..32].\n"
+    "\n"
+    "   -o, --out <file>\n"
+    "\n"
+    "       File to contain the new certificate. Specify '-' for stdout.\n"
```

Comment:
I don't think certificate is the right output. We should also explicitly state the output format of the base64 encoded verifier.

Reply:
doen